src/arch/x86/vm.c

```diff
@@ -123,6 +129,15 @@ int vm_arch_init_platform_device(vm_t *v)
     if (serial_init(&v->serial, &v->io_bus))
         return throw_err("Failed to init UART device");
     virtio_blk_init(&v->virtio_blk_dev);
+    vm_net_init(&v->virtio_net_dev);
+    return 0;
+}
+
+int vm_enable_net(vm_t *v){
```

Comment:
Is this an arch-dependent function? If not, the function should be in `vm.c` like `vm_load_diskimg` which calls `virtio_blk_init_pci()`.

Reply:
This is not an arch-dependent function. So I move the `vm_enable_net` to /src/vm.c.
`vm_enable_net` will initialize the tap device and then register the virtio-net device to bus.
```c
int vm_enable_net(vm_t *v){
    if(!virtio_net_init(&v->virtio_net_dev))
        return -1;
    virtio_net_init_pci(&v->virtio_net_dev,&v->pci,
                        &v->io_bus,&v->mmio_bus);
    return 0;
}
```